internal/venice-test-common/src/integrationtest/java/com/linkedin/davinci/notifier/LeaderDoomNotifier.java

```diff
@@ -0,0 +1,31 @@
+package com.linkedin.davinci.notifier;
+
+import static com.linkedin.venice.common.VeniceSystemStoreUtils.*;
+import static com.linkedin.venice.pushmonitor.ExecutionStatus.*;
+
+import com.linkedin.venice.pushmonitor.OfflinePushAccessor;
+
+
+/**
+ * A test only notifier to simulate ERROR in leader replica to test single leader replica failover scenario.
+ */
+public class LeaderDoomNotifier implements VeniceNotifier {
```

Comment:
BTW, should `LeaderDoomNotifier` extend `PartitionPushStatusNotifier` and just override `completed` function?
Maybe it is not a big deal for the test, but it may affect other logics, such as the partition status will become `COMPLETED` from nothing.